util/collections/machine_filters.go

```diff
@@ -214,3 +216,30 @@ func MatchesKubernetesVersion(kubernetesVersion string) Func {
 		return *machine.Spec.Version == kubernetesVersion
 	}
 }
+
+// ValidVersion returns a filter to find machine that have a non empty and valid version.
+func ValidVersion() Func {
```

Comment:
What about calling it
```suggestion
func WithVersion() Func {
```